mesonbuild/rewriter.py

```diff
@@ -857,16 +931,16 @@ def rel_source(src: str) -> str:
 
         elif cmd['operation'] == 'info':
             # T.List all sources in the target
-            src_list = []
-            for i in target.source_nodes:
-                for j in self.arg_list_from_node(i):
-                    if isinstance(j, StringNode):
-                        src_list += [j.value]
-            extra_files_list = []
-            for i in target.extra_files:
-                for j in self.arg_list_from_node(i):
-                    if isinstance(j, StringNode):
-                        extra_files_list += [j.value]
+
+            cwd = Path(os.getcwd())
+            source_root_abs = cwd / self.interpreter.source_root
+
+            src_list = self.interpreter.nodes_to_pretty_filelist(source_root_abs, target.subdir, target.source_nodes)
+            extra_files_list = self.interpreter.nodes_to_pretty_filelist(source_root_abs, target.subdir, [target.extra_files] if target.extra_files else [])
+
+            src_list = ['unknown' if isinstance(x, UnknownValue) else os.path.relpath(x, source_root_abs) for x in src_list]
```

Comment:
Use `mesonlib.relpath` to prevent exceptions if drives are different on Windows.

Reply:
Done